Merge 25.7 into trunk

jkmassel opened the pull request.

wpmobilebot committed:
Update draft release notes for Jetpack 25.7.

wpmobilebot committed:
Bump version number

wpmobilebot committed:
Update draft release notes for 25.7.

wpmobilebot committed:
Release Notes: add new section for next version (25.8)

iangmaia committed:
Fix `publish_release` lane (#21633)

wpmobilebot committed:
Freeze strings for translation

nbradbury committed:
Added android:autoVerify="true" to custom tabs service (#21635)

mokagio committed:
Download store metadata when generating beta build (#21623)

* Download store metadata when generating beta build

This should provide earlier feedback on the metadata translation, at
least for the portion that has already been translated at the time a
beta is made.

This is consistent with what we do in iOS, see
https://github.com/wordpress-mobile/WordPress-iOS/blob/b9a0deaf091739f33051f20cffdbe9a2097f645e/fastlane/lanes/release.rb#L219

* Remove call to push changes in download translated metadata lane

Automatically pushing from lanes other than the root-release lanes (code
freeze, new beta, finalize release, etc.) can result in undesired
pushes, like it just happened to me in
https://github.com/wordpress-mobile/WordPress-Android/commit/edcd40a47f4e4b80b049dc26f97aa335d7043659

dcalhoun committed:
fix: Address editor crashes from null listeners (#21657)

* fix: Address editor crashes from null listeners

Setting the listeners within `instantiateItem` appears problematic as
this only runs for the initial launch, but not for other events that
destroy and recreate fragment--e.g., screen rotation events.

To address this, GutenbergKit now follow the pattern of the Aztec and
GutenbergMobile editors, invoking the `initializeEditorFragment` method.
This method is described as "requesting dependency injection," which
feels like an odd statement. There is likely further opportunity to
refactor this architecture for simplicity.

* docs: Add release note

* docs: Remove `TODO` comment

Apparently the linter prohibits this statement...

nbradbury committed:
Fix ClassCastException caused by assuming editorFragment was a GutenbergEditorFragment when it could be a GutenbergKitEditorFragment (for 25.7) (#21667)

wpmobilebot committed:
Merge release_notes/25.7 into release/25.7 (#21691)

* Update 25.7 release notes

* Update WordPress `PlayStoreStrings.po` for version 25.7

* Update Jetpack `PlayStoreStrings.po` for version 25.7

---------

Co-authored-by: Oguz Kocer <oguz.kocer@automattic.com>

wpmobilebot committed:
Update translations

wpmobilebot committed:
Bump version number

wpmobilebot committed:
Update WordPress metadata translations for 25.7

wpmobilebot committed:
Update Jetpack metadata translations for 25.7

nbradbury committed:
Check the login mode before exiting when already logged in (#21695)

oguzkocer committed:
Manually bump version number for 25.7-rc-2

oguzkocer committed:
Update translations

oguzkocer committed:
Update WordPress metadata translations for 25.7

oguzkocer committed:
Update translations

oguzkocer committed:
Bump version number

oguzkocer committed:
Update WordPress metadata translations for 25.7

oguzkocer committed:
Update Jetpack metadata translations for 25.7

jkmassel committed:
Merge branch 'trunk' into merge/25.7-into-trunk

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR has the <code>Releases...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR has the <code>Releases</code> label: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
fix: Repair incorrect editor merge conflict resolution

A previous merge conflict resolution reverted changes that simplified
GutenbergKit by sharing the existing content change watcher.

See: https://github.com/wordpress-mobile/WordPress-Android/pull/21663

dcalhoun commented on fastlane/lanes/release.rb:
```diff
@@ -363,7 +363,7 @@
 
     publish_github_release(
-      repository: GITHUB_REPO,
+      repository: GHHELPER_REPO,
```
From reviewing https://github.com/wordpress-mobile/WordPress-Android/pull/21636, this change appears incorrect.

iangmaia replied:
This change was required by `25.7` ([see PR](https://github.com/wordpress-mobile/WordPress-Android/pull/21633/files)), but later `trunk` has been changed to use `GITHUB_REPO`, so we should avoid changing it back again.
```suggestion
      repository: GITHUB_REPO,
```

iangmaia commented on fastlane/lanes/release.rb:
```diff
@@ -613,7 +613,7 @@ def ensure_branch_does_not_exist!(branch_name)
   #
   def delete_remote_git_branch!(branch_name)
-    remove_branch_protection(repository: GITHUB_REPO, branch: branch_name)
+    remove_branch_protection(repository: GHHELPER_REPO, branch: branch_name)
```
Ditto
```suggestion
    remove_branch_protection(repository: GITHUB_REPO, branch: branch_name)
```

jkmassel committed:
Apply suggestions from code review

Co-authored-by: Ian G. Maia <iangmaia@users.noreply.github.com>

iangmaia reviewed (approved):
🚀
